scipy/special/_generate_pyx.py

```diff
@@ -606,26 +600,22 @@ def generate(self, all_loops):
         )
 
         for j, function in enumerate(loops):
-            toplevel += ("ufunc_%s_loops[%d] = <np.PyUFuncGenericFunction>%s\n" %
-                         (self.name, j, function))
+            toplevel += (f"ufunc_{self.name}_loops[{j}] = "
+                        f"<np.PyUFuncGenericFunction>{function}\n")
         for j, type in enumerate(types):
-            toplevel += "ufunc_%s_types[%d] = <char>%s\n" % (self.name, j, type)
+            toplevel += f"ufunc_{self.name}_types[{j}] = <char>{type}\n"
         for j, func in enumerate(funcs):
-            toplevel += "ufunc_%s_ptr[2*%d] = <void*>%s\n" % (
-                self.name, j, self.cython_func_name(func, specialized=True)
-            )
-            toplevel += "ufunc_%s_ptr[2*%d+1] = <void*>(<char*>\"%s\")\n" % (
-                self.name, j, self.name
-            )
+            toplevel += (f"ufunc_{self.name}_ptr[2*{j}] = <void*>"
+                        f"{self.cython_func_name(func, specialized=True)}\n")
+            toplevel += (f"ufunc_{self.name}_ptr[2*{j}+1] = <void*>"
+                        f"(<char*>\"{self.name}\")\n")
         for j, func in enumerate(funcs):
-            toplevel += "ufunc_%s_data[%d] = &ufunc_%s_ptr[2*%d]\n" % (
-                self.name, j, self.name, j)
-
-        toplevel += ('@ = np.PyUFunc_FromFuncAndData(ufunc_@_loops, '
-                     'ufunc_@_data, ufunc_@_types, %d, %d, %d, 0, '
-                     '"@", ufunc_@_doc, 0)\n' % (len(types)/(inarg_num+outarg_num),
-                                                 inarg_num, outarg_num)
-                     ).replace('@', self.name)
+            toplevel += f"ufunc_{self.name}_data[{j}] = &ufunc_{self.name}_ptr[2*{j}]\n"
+
+        toplevel += (f"@ = np.PyUFunc_FromFuncAndData(ufunc_@_loops, ufunc_@_data, "
+                    f"ufunc_@_types, {len(types)/(inarg_num + outarg_num)}, "
```

Comment:
the current error is due to this being a float rather than an integer, so perhaps your integer division with `//` was deliberate. I'll add an explicit cast to `int` here and see if that passes.